Car-service/src/main/java/carrest/CarServlet.java

```diff
@@ -0,0 +1,96 @@
+package carrest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.List;
+@WebServlet("/cars/*")
+public class CarServlet extends HttpServlet {
+    private final CarService carService = new CarService();
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String str = req.getPathInfo();
+        if (str == null) {
+            resp.setContentType("application/json");
+            List<Car> all = carService.getAll();
+            String json = mapper.writeValueAsString(all);
+            resp.getWriter().println(json);
+            System.out.println(all);
+        } else {
+            String[] parts = str.split("/");
+            String param = parts[1];
+            int carId = Integer.parseInt(param);
```

Comment:
this part can be moved to separate service or method since we can use it in other servlets (for example in other crud operation of other entities)

(it's related to doDelete method as well)